### ISSUE TITLE
../FPGA-Motion-Detector-master/FPGA-Motion-Detector-master/Vivado_HLS_MotionDetect/src_files/top.cpp:71:20: error: no matching function for call to 'Sum'     

### ISSUE THREAD

**Issue by kshitijzutshi** [russellbarnes/FPGA-Motion-Detector#1]
![image](https://user-images.githubusercontent.com/13203059/39542287-6fecf5f0-4e48-11e8-9615-5710eb9ece39.png)
I tried running the synthesis for the top.cpp file but there seems to be this error. 
img_diff is of type GRAY_IMAGE which is a hls::Mat type whereas result_pixel is a scalar type, i dont know how they can be assigned.. Can you please clarify?

**Comment by russellbarnes** [comment]
Hi, I no longer have access to the software to test this, but there are two things you could try. I think that this project was made for a newer version of HLS, something around 2016.2, so the newer version might resolve this call automatically? Also, since Sum is a template function, you could try filling in the template arguments manually: http://www.wiki.xilinx.com/HLS+Sum  hls::Sum<dest type, rows, cols, source type>(...)

**Comment by kshitijzutshi** [comment]
Yes, I have the latest version of Vivado HLS. About the template
suggesion., I have just started working with zynq fpga so I don't exactly
know how to rectify the error, I will try to work it out.
